### PR TITLE
feat(recipe): add RecipeInfo

### PR DESCRIPTION
**Pull request by mostaphaRoudsari**

### CODE REVIEW

File: queenbee/recipe/recipe.py [hunk 1]
```diff
@@ -680,3 +680,38 @@ def template_by_name(templates: List[Union[DAG, TemplateFunction]], name: str) -
...
 
         return res
+
+
+class RecipeInfo(BaseModel):
+    """Recipe information.
+
+    Recipe information only includes metadata, inputs and outputs of a Recipe.
+    This object is useful for creating user interface for Recipes.
+    """
```
Comment (AntoineDao) [inline queenbee/recipe/recipe.py:690]: Shouldn't there be a `type` key here?

**Review by AntoineDao — COMMENTED** [review]
I agree that we need an object like this. I wonder however if we might slightly change the naming in order to satisfy both the current C# requirements and possible future recipe builder requirements.

### Functional Design
Exposing some the recipe's `inputs` and `outputs` serves the purpose of making it usable as a functional component. In this sense it is very similar to a plugin's `Function` object.

I wonder if we might benefit from changing `RecipeInfo` to `RecipeFunction` and equally create a corresponding `PluginFunction` which contains plugin metadata + a single function's `io` information.This will be the same as what you propose here and in addition we will have a clear path to search and use functions (from recipes or plugins) when building recipes using a visual builder.

**Comment by AntoineDao** [comment]
After a call it seems we have agreed on the word `RecipeInterface` and this object should also contain a `source` key which is a url to where the recipe package can be downloaded.

File: queenbee/recipe/recipe.py (hunk 1 above)
Comment (mostaphaRoudsari) [reply queenbee/recipe/recipe.py:690]: Added!

**Comment by mostaphaRoudsari** [comment]
@AntoineDao, I changed it to RecipeInterface and added the source key. Ready for another round of review.

File: queenbee/recipe/recipe.py [hunk 2]
```diff
@@ -682,17 +682,24 @@ def template_by_name(templates: List[Union[DAG, TemplateFunction]], name: str) -
...
     This object is useful for creating user interface for Recipes.
     """
+    type: constr(regex='^RecipeInterface$') = 'RecipeInterface'
+
     metadata: MetaData = Field(
         ...,
         description='Recipe metadata information.'
     )
 
+    source: str = Field(
```
Comment (AntoineDao) [inline queenbee/recipe/recipe.py:698]: Reckon we can use the [`AnyUrl`](https://pydantic-docs.helpmanual.io/usage/types/#urls) type from Pydantic here maybe?

**Review by AntoineDao — APPROVED** [review]
LGTM! Small detail in comments but we can merge as is.

File: queenbee/recipe/recipe.py (hunk 2 above)
Comment (mostaphaRoudsari) [reply queenbee/recipe/recipe.py:698]: I like using a more rigid type but AnyUrl doesn't allow `file:///` format.

**Pull request merged**